### PR TITLE
Macros in Variants

### PR DESCRIPTION
**Pull request by pmbittner**
This pull-requests adds support for including preprocessor directives in generated variants. Upon variant generation, users can choose to in- or exclude preprocessor directives from variants with the new option `withMacros` in the `VariantGenerationOptions` (previously, preprocessor directives where always excluded). We need this feature to treat variants as "sub"-product lines for certain analyses where feature traceability has to be given internally (in the file) and not only externally (as we already do in the ground truth files).

### CODE REVIEW

**Comment by pmbittner** [comment]
Current Status: Tests succeed
Yet, tests where build without preprocessor directives in variants in mind, so further testing would be good (e.g., to see if the generated preprocessor directives are correct).

File: src/test/java/org/variantsync/vevos/simulation/VariantGenerationTest.java [hunk 1]
```diff
@@ -111,6 +111,7 @@ public boolean generate(final List<Variant> variantsToTest, final boolean writeC
     @BeforeClass
     public static void setupStatic() {
         VEVOS.Initialize();
+        Result.HARD_CRASH_ON_TRY = true;
```
Comment (AlexanderSchultheiss) [inline src/test/java/org/variantsync/vevos/simulation/VariantGenerationTest.java:114]: I get the following error on this line: `java: cannot assign a value to final variable HARD_CRASH_ON_TRY`
I executed the ./build-local-repo.sh before, so I should have the up-to-date version of the _functjonal_ library.

**Review by AlexanderSchultheiss — COMMENTED** [review]
Looks good. I only have one minor issue with the setting of a variable.

File: src/test/java/org/variantsync/vevos/simulation/VariantGenerationTest.java (hunk 1 above)
Comment (pmbittner) [reply src/test/java/org/variantsync/vevos/simulation/VariantGenerationTest.java:114]: Hmmm, but on the main branch of Functjonal, this field is not final: https://github.com/VariantSync/Functjonal/blob/main/src/main/java/org/variantsync/functjonal/Result.java

Are there any caching issues in maven?

Comment (pmbittner) [reply src/test/java/org/variantsync/vevos/simulation/VariantGenerationTest.java:114]: The jar has to be updated. I will do that.

Comment (pmbittner) [reply src/test/java/org/variantsync/vevos/simulation/VariantGenerationTest.java:114]: I updated the local maven repo and jar file now. Could you check if the error is gone now for you?

Comment (AlexanderSchultheiss) [reply src/test/java/org/variantsync/vevos/simulation/VariantGenerationTest.java:114]: Yes, it is working now.

**Review by AlexanderSchultheiss — APPROVED** [review]

**Comment by AlexanderSchultheiss** [comment]
Merging into new branch 'next-release', which will later be merged into main

**Pull request merged**